### PR TITLE
Asserts in Pauli tests 

### PR DESCRIPTION
**Pull request by lbello**
## Description
The `test_qi.TestPauli` had a lot of logging but now so many asserts. On top of that, too much random. That's not good (see #205). So I added some asserts (mostly, for the things that were logged) and I removed random inputs (with the exception of `random_pauli`, where the invariants are checked.) This is a partial fix to #205.

## How Has This Been Tested?
`make test`

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by lbello** [comment]
I am passing lint locally, with `make lint`.

File: test/python/test_qi.py [hunk 1]
```diff
@@ -66,9 +66,9 @@ def test_vectorize(self):
         row = [1, 2, 3, 4]
         paul = [5, 5, -1j, -3]
         test_pass = np.linalg.norm(vectorize(mat) - col) == 0 and \
```
[the remark this answers is not in the document]
Comment (lbello) [reply test/python/test_qi.py]: This part was touch when I ran "Reformat code" in PyCharm. I fixed it following you advice.

File: test/python/test_qi.py [hunk 2]
```diff
@@ -129,101 +129,183 @@ def test_concurrence(self):
...
+        self.log.info(self.p3.to_matrix())
+        m = '''[[ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.-1.j  0.+0.j]
+ [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+1.j]
+ [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.-1.j  0.+0.j  0.+0.j  0.+0.j]
+ [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+1.j  0.+0.j  0.+0.j]
+ [ 0.+0.j  0.+0.j  0.+1.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+ [ 0.+0.j  0.-0.j  0.+0.j  0.-1.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+ [ 0.+1.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+ [ 0.+0.j  0.-1.j  0.+0.j  0.-0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]]'''
+        self.assertMultiLineEqual(str(self.p3.to_matrix()), m)
```
[the remark this answers is not in the document]
Comment (lbello) [reply test/python/test_qi.py]: I fixed the issue by comparing equality between matrices, instead of strings.

**Comment by diego-plan9** [comment]
Thanks @lbello - cleaning up the tests and advancing towards #205 is much appreciated! We should look at the fixed failing test (probably adding a tests that exerts just the `pauli.to_matrix()` function) at some point, as it seems to indicate different behaviour under different conditions (most likely due to different versions of some dependencies), but it can be done in another PR and with less priority.

**Pull request merged**